### ISSUE TITLE
siac upload command doesn't actually upload

### ISSUE THREAD

**Issue by seveibar** [NebulousLabs/Sia#263]
I haven't tested, but I was looking at this snippet...

```go
func fileuploadcmd(file, nickname, pieces string) {
	err := callAPI(fmt.Sprintf("/file/upload?file=%s&nickname=%s&pieces=%s", file, nickname, pieces))
	if err != nil {
		fmt.Println("Could not upload file:", err)
		return
	}
	fmt.Println("Uploaded", file, "as", nickname)
}
```

And I don't see it actually doing any uploading i.e. a file needs to be set as a form value.

I'll probably know how to fix this soon because I'm doing the same thing with the API tests.

Alternatively (and this is really not a bad idea) we could add a path variable or have a /file/uploadpath handler to avoid unnecessary overhead.

**Comment by lukechampine** [comment]
good catch. really I think using a filename should be the default, and a separate call could be used for "actual" uploading.
Alternatively, we could do both in the same call...
```go
func (d *daemon) fileUploadHandler(w http.ResponseWriter, req *http.Request) {
	// ...
	if req.FormValue("filename") != "" {
		// process filename
	} else {
		// process upload
		file, _, err := req.FormFile("file")
	}
	//...
}
```
this might be the best option if we want to patch this quickly.

**Comment by seveibar** [comment]
fixed with #264
